src/libecalc/common/utils/rates.py

```diff
@@ -156,6 +156,23 @@ def convert_none_to_nan(cls, v: float, field: ModelField) -> TimeSeriesValue:
             return math.nan
         return v
 
+    @validator("values", pre=True)
+    def timesteps_values_one_to_one(cls, v: list, values, field: ModelField):
+        nr_timesteps = len(values["timesteps"])
+        nr_values = len(v)
+
+        if not cls.__name__ == TimeSeriesVolumes.__name__:
+            if nr_timesteps != nr_values:
+                if all(math.isnan(i) for i in v):
+                    # TODO: This should probably be solved another place. Temporary solution to make things run
```

Comment:
yes, this is a symptom that nr of timesteps are off...for some reason. It shouldnt be...right? It should be fixed where error arises, to make sure we have the same nr of values all the time...

Reply:
This is most likely a bug, it happens in `EnergyModelBaseResult.extend()`, related to handling of None etc. Will be reported as separate issue.